src/Util.cpp

```diff
@@ -85,10 +85,18 @@ bool saveFile(const QString &path, const QString &content, const QString &head,
     return true;
 }
 
-QString readFile(const QString &path, const QString &head, MessageLogger *log)
+QString readFile(const QString &path, const QString &head, MessageLogger *log, bool notExistWarning)
 {
     if (!QFile::exists(path))
+    {
+        if (notExistWarning)
+        {
+            if (log != nullptr)
+                log->warn(head, QString("The file [%1] does not exist").arg(path));
+            LOG_WARN(QString("The file [%1] does not exist").arg(path));
```

Comment:
Not required and duplicate logs will be produced log->warn("..."); always writes the message to the log file as well as message box of UI. Maybe you need to use 

```LOG_WARN_IF(log == nullptr, QString("The file [%1] does not exists).args(path));```
or an else statement before `LOG_WARN`, so that two same messages does not get logged.

Reply:
`log->warn` is logged as information, but we need a warning in the log file.

Reply:
This change was brought by you where all Editor messages box messages are by default in INFO section of Log Files. what about duplication of Logs? Why put it to INFO? 

This is same as putting LOG_INFO and LOG_WARN both inside messagelogger.warn()? Also this is Util function that can be called multiple times so it can SPAM the Log file with duplicate logs.

A possible alternative will be to add a prefix "MessageLogger" before all LOGS that are written with log.info/warn/error. So,

```cpp

log->warn("Something is missing");

```

Internally writes:

`LOG_WARN("[MessageLogger]" << "Something is missing ");`

Reply:
Because not all warnings for the user should be a warning in the log file, for example, the compilation warnings.

Reply:
> Also this is Util function that can be called multiple times so it can SPAM the Log file with duplicate logs.

1. Now this is used only here (when adding paris test cases from files).
2. This is logged only when the file doesn't exist unexpectedly.

So the double logging here won't spam.

Reply:
Okay then agreed!